zeroeq/http/server.cpp

```diff
@@ -68,58 +98,71 @@ class Server::Impl : public detail::Sender
         initURI();
     }
 
-    bool subscribe( servus::Serializable& serializable )
+    bool handlePUT( servus::Serializable& serializable )
     {
         const auto func = [&serializable]( const std::string& json )
             { return serializable.fromJSON( json ); };
-        return subscribe( serializable.getTypeName(), func );
+        return handlePUT( serializable.getTypeName(), serializable.getSchema(),
+                          func );
     }
 
-    bool subscribe( std::string event, const PUTPayloadFunc& func )
+    bool handlePUT( std::string event, const std::string& schema,
+                    const PUTPayloadFunc& func )
     {
-        event = _toLower( event );
-        if( _subscriptions.count( event ) != 0 )
+        event = _toLower( _replaceAll( event, "/", "::" ));
+        if( _put.count( event ) != 0 )
             return false;
 
-        _subscriptions[ event ] = func;
+        _put[ event ] = func;
+        if( !schema.empty( ))
+            _schemas[ event ] = schema;
         return true;
     }
 
-    bool unsubscribe( const servus::Serializable& serializable )
+    bool removePUT( const servus::Serializable& serializable )
     {
-        return unsubscribe( serializable.getTypeName( ));
+        return removePUT( serializable.getTypeName( ));
     }
 
-    bool unsubscribe( const std::string& event )
+    bool removePUT( std::string event )
     {
-        return _subscriptions.erase(
-            _toLower( event )) != 0;
+        event = _toLower( _replaceAll( event, "/", "::" ));
+        if( _get.count( event ) == 0 )
+            _schemas.erase( event );
+        return _put.erase( event ) != 0;
     }
 
-    bool register_( servus::Serializable& serializable )
+    bool handleGET( servus::Serializable& serializable )
     {
-        const auto func = [&serializable]() { return serializable.toJSON(); };
-        return register_( serializable.getTypeName(), func );
+        const auto func = [&serializable] { return serializable.toJSON(); };
+        return handleGET( serializable.getTypeName(), serializable.getSchema(),
+                          func );
     }
 
-    bool register_( std::string event, const GETFunc& func )
+    bool handleGET( std::string event, const std::string& schema,
+                    const GETFunc& func )
     {
-        event = _toLower( event );
-        if( _registrations.count( event ) != 0 )
+        event = _toLower( _replaceAll( event, "/", "::" ));
+        if( _get.count( event ) != 0 )
```

Comment:
I don't know if it's relevant, but MSVC used to whine about a potential performance bug in the implicit int to bool conversion if you do that.